src/rust/ensogl/lib/core/src/display/scene/dom.rs

```diff
@@ -193,22 +113,104 @@ impl DomScene {
     pub fn update_view_projection(&self, camera:&Camera2d) {
         if self.children_number() == 0 { return }
 
-        let trans_cam  = camera.transform_matrix().try_inverse();
-        let trans_cam  = trans_cam.expect("Camera's matrix is not invertible.");
-        let trans_cam  = trans_cam.map(eps);
-        let trans_cam  = inverse_y_translation(trans_cam);
-        let half_dim   = camera.screen().height / 2.0;
-        let fovy_slope = camera.half_fovy_slope();
-        let near       = half_dim / fovy_slope;
+
+        // === Reset configuration ===
+
+        self.data.view_projection_dom.set_style_or_panic("transform", "");
+        self.data.view_projection_dom.set_style_or_panic("left", "0px");
+        self.data.view_projection_dom.set_style_or_panic("top", "0px");
+        self.data.dom.set_style_or_panic("perspective", "");
+
+
+        let view_matrix = camera.view_matrix();
+        let view_matrix = view_matrix.map(eps);
+        // In CSS, the y axis points downwards. (In EnsoGL upwards)
+        let mut trans_cam = flip_y_axis(view_matrix);
 
         match camera.projection() {
             Projection::Perspective{..} => {
-                js::setup_perspective(&self.data.dom,&near.into());
-                setup_camera_perspective(&self.data.view_projection_dom,near,&trans_cam);
+                // In order to achieve a better visual quality when the camera is in fron of the z=0
+                // plane, we simulate translation in the z direction by scaling the scene around
+                // the origin, as discussed at https://github.com/enso-org/ide/pull/1465.
+                // In that situation, we also simulate x and y translation through `top` and `left`.
+
+                let x_index = (0, 3);
+                let y_index = (1, 3);
+                let z_index = (2, 3);
+                let target_x = *trans_cam.index(x_index);
+                let target_y = *trans_cam.index(y_index);
+                let target_z = *trans_cam.index(z_index);
+
+                // Put the camera in front of the origin and adjust the field of view, such that one
+                // px unit at z=0 maps to one px on screen.
+                self.data.dom.set_style_or_panic("perspective", format!("{}px", camera.z_zoom_1()));
```

Comment:
I think we also have `set_style_or_warn`, which might be safer to use as it doesn't crash the IDE if something goes wrong.